### ISSUE TITLE
Error importing LinearOperator on Google Colab

### ISSUE THREAD

**Issue by tito21** [PyLops/pylops#201]
After release 1.12.1 importing LinearOperator using: `from pylops import LinearOperator` throws the following error in Google Colab

```
ImportError                               Traceback (most recent call last)
<ipython-input-4-70b12ce7c165> in <module>()
      1 import numpy as np
----> 2 from pylops import LinearOperator
      3 
      4 
      5 class UndersampleObesrver(LinearOperator):

4 frames
/usr/local/lib/python3.6/dist-packages/pylops/__init__.py in <module>()
----> 1 from .LinearOperator import LinearOperator
      2 from .basicoperators import Regression
      3 from .basicoperators import LinearRegression
      4 from .basicoperators import MatrixMult
      5 from .basicoperators import Identity

/usr/local/lib/python3.6/dist-packages/pylops/LinearOperator.py in <module>()
     15 from scipy.sparse import csr_matrix
     16 
---> 17 from pylops.utils.backend import get_array_module, get_module, get_sparse_eye
     18 from pylops.optimization.solver import cgls
     19 

/usr/local/lib/python3.6/dist-packages/pylops/utils/__init__.py in <module>()
      1 from .utils import Report
----> 2 from .dottest import dottest
      3 from .deps import *
      4 from .backend import *

/usr/local/lib/python3.6/dist-packages/pylops/utils/dottest.py in <module>()
      1 import numpy as np
----> 2 from pylops.utils.backend import get_module
      3 
      4 
      5 def dottest(Op, nr, nc, tol=1e-6, complexflag=0, raiseerror=True, verb=False,

/usr/local/lib/python3.6/dist-packages/pylops/utils/backend.py in <module>()
      8     import cupy as cp
      9     import cupyx
---> 10     from cupyx.scipy.linalg import block_diag as cp_block_diag
     11     from cupyx.scipy.linalg import toeplitz as cp_toeplitz
     12     from cupyx.scipy.sparse import csc_matrix as cp_csc_matrix

ImportError: cannot import name 'block_diag'
```
Seems like some dependency error with cupy. I tested locally installed version via pip and it worked. Downgrading to version 1.11.1 fixes the error

**Comment by mrava87** [comment]
Hey, you are correct. I have experienced the same and it is due to cupy being too old on colab.

I have a quick fix solution, will update the doc today and point you to it (which basically requires adding !pip install cupy-cuda101==8.0.0rc1 --upgrade in a cell before you start your imports)

Ultimately I plan to make a patch where the version of cupy is also checked and if it is available below a certain version it is assumed not to be there... the reason we can’t add cupy directly to the dependencies is because PyLops works just fine without but then we have to do internal checks ourselves as we don’t require it to be there :)

**Comment by mrava87** [comment]
If you do this, so not downgrade PyLops, you can now use cupy arrays and get massive to good (depending on which operator your use and if your problem is more or less suitors to leverage the power of gpus) : 
https://pylops.readthedocs.io/en/latest/gpu.html

See here for some examples https://github.com/PyLops/pylops_notebooks/tree/master/developement-cupy and more specifically here on how to setup colab to also use cusignal powered operators https://github.com/PyLops/pylops_notebooks/blob/master/developement-cupy/SignalProcessing.ipynb

Hope it helps :)

**Comment by tito21** [comment]
Hi, thanks for the answer. 
Upgrading cupy solve the issue. Still I think I would stick with previous version 1.11 because I am benchmarking CPU implementation vs GPU using [pylops-gpu](https://github.com/PyLops/pylops-gpu) with pytorch. I don't think I can manage to change strategies right now. But I will be checking back in the future.

**Comment by mrava87** [comment]
Great :) this a very fresh addition and so far I think has more potential that pylops-gpu if you are purely interested in speeding up the operators with gpus (and also much easier to integrate as cupy and numpy are almost identical in their API. Note that you can use the new version 1.12, if you just use numpy arrays it will behave identical to 1.11, the only difference is that if you now pass cupy arrays it will automatically switch to cupy and run your computations on the gpu :)

Pylops-gpu is great (although I have had not much time to work on it lately) especially if you want to mix pytorch ML operators with pylops linear operators... it is a bit smaller than pylops as some stuff were not available when i worked on it (e.g. complex numbers and fourier transforms, but they are slowly coming to pytoch) :)

**Comment by mrava87** [comment]
Hi, this is fixed with the new version 1.13.0, by simply using the env flags in case cupy/cusignal are available but their version is too old (like in Colab at the moment)
